high_card.rb

```diff
@@ -1,5 +1,86 @@
-# Use these two arrays to generate a deck of cards.
-ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
-suits = [ "hearts", "spades", "clubs", "diamonds" ]
+def getPlayers 
+    players, break_word = {}, 'play'
+    while true
+        print "#{players.size} players so far.\n" +
+              "Enter a player name, or type '#{break_word}': "
+        player = gets.chomp
+        break if player == break_word
+        # TODO: store number of wins as a value for each player
+        # to be displayed at the end of each game series
+        players.update(player => nil) if player != "" and not
+                                         players.key?(player)
+    end
+    players
+end
 
-players = []
+def deal players, deck
+    shuffled_deck = deck.shuffle
+    players.keys.map.with_index do |player, i|
+        {name: player, card: shuffled_deck[i][0], suit: shuffled_deck[i][1]}
+    end
+end
+
+def getWinner hands, ranks
+    winning_card = hands.map {|hand| ranks[hand[:card]]}.max
+    hands.select {|hand| hand if ranks[hand[:card]] == winning_card}
+end
+
+def declareWinner winner
+    if winner.size > 1
+        tie_message = "It's a tie between "
+        hand_messages = winner.map do |hand|
+            "#{hand[:name]} with a #{hand[:card]} of #{hand[:suit]}" 
+        end
+        puts "#{tie_message} " +
+             "#{hand_messages.first(hand_messages.size-1).join(', ')}" +
+             ", and #{hand_messages[-1]}"
+        return false
+    else
+        puts "#{winner[0][:name]} wins with a " +
+             "#{winner[0][:card]} of #{winner[0][:suit]}"
+    end
+    true
+end
+
+RANKS = {
+    "A" => 1,
+    2 => 2,
+    3 => 3,
+    4 => 4,
+    5 => 5,
+    6 => 6,
+    7 => 7,
+    8 => 8,
+    9 => 9,
+    10 => 10,
+    "J" => 11,
+    "Q" => 12,
+    "K" => 13
+}
+SUITS = [ "hearts", "spades", "clubs", "diamonds" ]
+MIN_PLAYERS, MAX_PLAYERS = 2, RANKS.size * SUITS.size
+
```

Comment:
I figured that these were parameters/rules of the game that would never change throughout the program, so it might be best to declare them as constants.